src/openvic-simulation/types/Sliders.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+  // TODO: make struct for tariff slider, unlike other sliders, you can have negative values
+  struct slider_value_t {
+  friend struct  SliderManager;
+  private:
+      int PROPERTY(min);
+      int PROPERTY(max);
+      int PROPERTY(value);
+
+  public:
+      constexpr slider_value_t(int value, int min = 0, int max = 100) {
+        // You *can* actually have min > max in Victoria 2
+        // Such a situation will result in only being able to be move between the max and min value. 
+        // This logic replicates this "feature"
+        if (value > max) {
+          value = min;
+        }
+        if (value < min) {
+          value = max;
+        }
+      }
+  };
+
+class SliderManager {
+  public:
+    void set_slider_value(slider_value_t& slider, int new_value) {
```

Comment:
Why not make this a method of the slider itself?

Reply:
Yeah there's no need for a `SliderManager` at all, instead the struct itself should have these setter functions:
- `void set_value(int_type new_value)` - clamping `new_value` between min and max before setting it (any logic about the `max < min` case should happen in this function too)
- `void set_bounds(int_type new_min, int_type new_max)` - clamping min and max between the final/constant min and max values, then also updating the value in case it's no longer in the allowed range (you could simply call `set_value(value)`)